bsp/src/org/jetbrains/bsp/project/resolver/BspResolverLogic.scala

```diff
@@ -395,17 +396,17 @@ private[resolver] object BspResolverLogic {
     }
 
   private[resolver] def projectNode(workspace: File,
+                                    jvmEnvironment: Option[WorkspaceJvmEnvironmentResult],
                                     projectModules: ProjectModules): DataNode[ProjectData] = {
 
     val projectRootPath = workspace.getCanonicalPath
     val moduleFilesDirectoryPath = moduleFilesDirectory(workspace).getCanonicalPath
     val projectRoot = new File(projectRootPath)
     val projectData = new ProjectData(BSP.ProjectSystemId, projectRoot.getName, projectRootPath, projectRootPath)
     val projectNode = new DataNode[ProjectData](ProjectKeys.PROJECT, projectData, null)
-
-    // TODO have a way to set project jdk before/during import
-    val bspProjectData = new DataNode[BspProjectData](BspProjectData.Key, BspProjectData(None), projectNode)
-    projectNode.addChild(bspProjectData)
+    val bspProjectData = jvmEnvironment.map { env =>
```

Comment:
This makes creation of project dependent on existens of a jvmEnvironment. But the existence of such an endpoint isn't and shouldn't be mandatory